examples/samplers/simulated_annealing_sampler.py

```diff
@@ -0,0 +1,146 @@
+"""
+Optuna example that implements a user-defined relative sampler based on Simulated Annealing
+algorithm. Please refer to https://en.wikipedia.org/wiki/Simulated_annealing for Simulated
+Annealing itself.
+
+Note that this implementation isn't intended to be used for production purposes and
+has the following limitations:
+- The sampler only supports `UniformDistribution` (i.e., `Trial.suggest_uniform` method).
+- The implementation prioritizes simplicity over optimization efficiency.
+
+You can run this example as follows:
+    $ python simulated_annealing_sampler.py
+
+"""
+
+import numpy as np
+
+import optuna
+from optuna import distributions
+from optuna.samplers import BaseSampler
+from optuna import structs
+
+if optuna.types.TYPE_CHECKING:
```

Comment:
Since this is an example, we can get rid of type hint to make the script simpler.

Reply:
Sounds good. Removed https://github.com/pfnet/optuna/pull/446/commits/65e861687a793334d4273dc5c6be29c894a3b210.